service/api_task.go

```diff
@@ -310,14 +310,25 @@ func (as *APIServer) EndTask(w http.ResponseWriter, r *http.Request) {
 		endTaskResp.ShouldExit = true
 	}
 
-	grip.Info(message.Fields{
+	msg := message.Fields{
 		"message":     "Successfully marked task as finished",
 		"task_id":     t.Id,
 		"execution":   t.Execution,
 		"operation":   "mark end",
 		"duration":    time.Since(finishTime),
 		"should_exit": endTaskResp.ShouldExit,
-	})
+	}
+
+	if t.IsPartOfDisplay() {
+		dt, err := t.GetDisplayTask()
+		if err != nil {
+			grip.Error(errors.Wrapf(err, "can't get display task for task %s", t.Id))
+		} else {
+			msg["display_task_id"] = dt.Id
```

Comment:
t.IsPartOfDisplay saves the display task to t if it is found, so rather than call GetDisplayTask / error checking again you can just directly reference t.DisplayTask.Id here (same comments below)